Musl: error[E0425]: cannot find function `close_range` in crate `libc`

compiling on musl fails with error:

```Rust
error[E0425]: cannot find function `close_range` in crate `libc`
  --> src/main.rs:37:28
   |
37 |     let _ = unsafe { libc::close_range(3, libc::c_uint::MAX, 0) };
   |                            ^^^^^^^^^^^ not found in `libc`

For more information about this error, try `rustc --explain E0425`.
error: could not compile `aardvark-dns` (bin "aardvark-dns") due to 1 previous error
```

due to `close_rang` (https://github.com/containers/aardvark-dns/commit/2599cdd1113501d2c3618f685151a3f3b38a1ada) being available only on glibc (https://github.com/rust-lang/libc/pull/3373).

Downstream bug: https://bugs.gentoo.org/980641

--- a/main/aardvark-dns/patches/close_range-syscall.patch
+++ b/main/aardvark-dns/patches/close_range-syscall.patch
@@ -1,0 +1,43 @@
+From 9e73fd8ddde844d755f3f992ca9cd7bb0502d87e Mon Sep 17 00:00:00 2001
+From: Pepper Gray <hello@peppergray.xyz>
+Date: Tue, 11 Aug 2026 00:03:34 +0200
+Subject: [PATCH] add syscall for close_range for non-glibc targets
+
+compiling on musl fails with error:
+```
+error[E0425]: cannot find function `close_range` in crate `libc`
+```
+
+use syscall for targets which do not provide a `close-range`-wrapper
+
+Closes: #730
+Signed-off-by: Pepper Gray <hello@peppergray.xyz>
+---
+ src/main.rs | 12 +++++++++++-
+ 1 file changed, 11 insertions(+), 1 deletion(-)
+
+diff --git a/src/main.rs b/src/main.rs
+index f3af0d6a..c20761e1 100644
+--- a/src/main.rs
++++ b/src/main.rs
+@@ -32,9 +32,19 @@ enum SubCommand {
+     Version(version::Version),
+ }
+ 
++#[cfg(any(all(target_os = "linux", target_env = "gnu"), target_os = "freebsd",))]
++unsafe fn close_range(first: libc::c_uint, last: libc::c_uint, flags: libc::c_int) -> libc::c_int {
++    unsafe { libc::close_range(first, last, flags) }
++}
++
++#[cfg(all(target_os = "linux", not(target_env = "gnu"),))]
++unsafe fn close_range(first: libc::c_uint, last: libc::c_uint, flags: libc::c_int) -> libc::c_int {
++    unsafe { libc::syscall(libc::SYS_close_range, first, last, flags) as libc::c_int }
++}
++
+ fn main() {
+     // Close all fds we inherited from our parent process, we only need stdio.
+-    let _ = unsafe { libc::close_range(3, libc::c_uint::MAX, 0) };
++    let _ = unsafe { close_range(3, libc::c_uint::MAX, 0) };
+ 
+     let formatter = Formatter3164 {
+         facility: Facility::LOG_USER,

--- a/main/aardvark-dns/template.py
+++ b/main/aardvark-dns/template.py
@@ -1,5 +1,5 @@
 pkgname = "aardvark-dns"
-pkgver = "2.0.0"
+pkgver = "2.1.0"
 pkgrel = 0
 build_style = "cargo"
 hostmakedepends = ["cargo-auditable"]
@@ -8,7 +8,7 @@ pkgdesc = "Authoritative DNS server for A/AAAA container records"
 license = "Apache-2.0"
 url = "https://github.com/containers/aardvark-dns"
 source = f"{url}/archive/v{pkgver}.tar.gz"
-sha256 = "d3f5d6b3be3c2d80e8257fb9467e34ff104f299474427979454034dca6dc88cc"
+sha256 = "daf871488603e659b0501224cf0731ac317809b1d1701fc061cb4f6ae39a894f"
 
 
 def install(self):

